lib/ansible/cli/vault.py

```diff
@@ -142,6 +142,13 @@ def post_process_args(self, options):
             if '-' in options.args or not options.args or options.encrypt_string_stdin_name:
                 self.encrypt_string_read_stdin = True
 
+                # If we're going to read 1 arg from stdin implicitly (no command line args),
+                # set --stdin-name to the first --name if it wasn't provided
+                if not options.encrypt_string_stdin_name and not options.args and options.encrypt_string_names:
+                    options.encrypt_string_stdin_name = options.encrypt_string_names[0]
+                    if len(options.encrypt_string_names) > 1:
+                        display.display('The number of --name options do not match the number of args.', stderr=True)
```

Comment:
The more I think about it, maybe giving an error here would be better. Then at least `--name` would not be ignored for this cornercase, without conflating `--name` and `--stdin-name`. They are completely separate.